pynucastro/networks/tests/test_compare_cxx_and_python.py

```diff
@@ -0,0 +1,78 @@
+# this creates the same network as SimpleCxxNetwork and PythonNetwork
+# and have them both compute dY/dt and compares to make sure that they
+# agree.  Note: screening is not considered.
+
+import os
+import re
+import subprocess
+from pathlib import Path
+
+import numpy as np
+import pytest
+from pytest import approx
+
+import pynucastro as pyna
+
+
+class TestNetworkCompare:
+
+    @pytest.fixture(scope="class")
+    def lib(self, reaclib_library):
+        nuc = ["p", "he4", "c12", "o16", "ne20", "na23", "mg24"]
+        lib = reaclib_library.linking_nuclei(nuc)
+        return lib
+
+    def test_compare(self, lib):
+
+        test_path = Path("_test_compare/")
+
+        test_path.mkdir(parents=True, exist_ok=True)
+        os.chdir(test_path)
+
+        # C++
+
+        cxx_net = pyna.SimpleCxxNetwork(libraries=[lib])
+        cxx_net.write_network()
+
+        subprocess.run("make DISABLE_SCREENING=TRUE", capture_output=False,
+                       shell=True, check=True)
+
+        cp = subprocess.run("./main", capture_output=True,
+                            shell=True, check=True, text=True)
+        stdout = cp.stdout
+
+        ydot_re = re.compile(r"(Ydot)\((\w*)\)(\s+)(=)(\s+)([\d\-e\+.]*)",
+                             re.IGNORECASE | re.DOTALL)
+
+        ydots_cxx = {}
+        for line in stdout.split("\n"):
+            if match := ydot_re.search(line.strip()):
+                ydots_cxx[match.group(2)] = float(match.group(6))
+
+        os.chdir("../")
```

Comment:
This won't be run if either of the subprocesses fail, which breaks other tests

Reply:
It affects everything. You can avoid the chdirs by adding `odir=test_path` to `write_network()` and `cwd=test_path` to `subprocess.run()`

Reply:
ah... that's a good solution

Reply:
fixed

Reply:
this line also needs to be removed, as well as `import os` at the top

Reply:
done